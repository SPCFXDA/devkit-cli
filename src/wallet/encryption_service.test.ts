import { snapshotTest } from 'cliffy/testing'
import { ansi } from 'cliffy/ansi'
import { EncryptionService } from './encryption_service.ts'

// import { assertEquals, assertRejects } from "@std/assert";

const mockGetRandomValues = (typedArray: Uint8Array) => {
	return typedArray
}

await snapshotTest({
	name: 'EncryptionService - encrypt and decrypt mnemonic successfully',
	meta: import.meta,
	stdin: ansi
		.text('password\n')
		.text('password\n')
		.toArray(),
	async fn() {
		const encryptionService = new EncryptionService()
		const mnemonic = 'test test test test test test test test test test test junk'
		encryptionService.getRandomValues = mockGetRandomValues
		const encryptedMnemonic = await encryptionService.encryptMnemonic(mnemonic)

		console.log('encrypted:', encryptedMnemonic)
		console.log('decrypted', await encryptionService.decryptMnemonic(encryptedMnemonic))
	},
})
