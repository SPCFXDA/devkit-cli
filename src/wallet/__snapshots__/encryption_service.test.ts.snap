export const snapshot = {};

snapshot[`EncryptionService - encrypt and decrypt mnemonic successfully 1`] = `
stdout:
"Uint8Array(12) [
  0, 0, 0, 0, 0,
  0, 0, 0, 0, 0,
  0, 0
] Uint8Array(16) [
  0, 0, 0, 0, 0, 0,
  0, 0, 0, 0, 0, 0,
  0, 0, 0, 0
]
? Enter encryption password to secure your mnemonic. › \\x1b[56G\\x1b[G\\x1b[0J
? Enter encryption password to secure your mnemonic. › ********\\x1b[64G\\x1b[G\\x1b[0J
? Enter encryption password to secure your mnemonic. › ********
\\x1b[?25h\\x1b[?25h? Password:  › \\x1b[16G\\x1b[G\\x1b[0J
? Password:  › ********\\x1b[24G\\x1b[G\\x1b[0J
? Password:  › ********
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
